ITechArt.FlightBookingsAPI.Services/Services/UsersService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using ITechArt.FlightBookingsAPI.Domain.Errors;
+using ITechArt.FlightBookingsAPI.Domain.Models;
+using ITechArt.FlightBookingsAPI.Services.Interfaces;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.IdentityModel.Tokens;
+
+namespace ITechArt.FlightBookingsAPI.Services.Services;
+
+public class UsersService : IUsersService
+{
+    private const string UserRole = "User";
+    private const string AdminRole = "Admin";
+
+    private readonly UserManager<User> _userManager;
+
+    public UsersService(UserManager<User> userManager)
+    {
+        _userManager = userManager;
+    }
+    
+    public async Task Register(User user, string password)
+    {
+        var result = await _userManager.CreateAsync(user, password);
+        if (!result.Succeeded)
+        {
+            throw new BadRequestException(result.Errors.First().Description);
+        }
+
+        result = await _userManager.AddToRoleAsync(user, UserRole);
+        if (!result.Succeeded)
+        {
+            throw new BadRequestException(result.Errors.First().Description);
+        }
+    }
+
+    public async Task<string> Login(string username, string password, string securityKey, string issuer)
+    {
+        var user = await _userManager.FindByNameAsync(username);
+        if (user is null || !await _userManager.CheckPasswordAsync(user, password))
+        {
+            throw new KeyNotFoundException("Incorrect login/password");
```

Comment:
KeyNotFoundException is a wrong exception in this case as this exception should be used in cases when we can't retrieve an object from collection by given key, and in this case we're doing pretty different thing - we may check existing user but try a wrong password. I'd say that the most appropriate exception in this case is ObjectNotFoundException, but as it requires Entity dll then I'd prefer to use ArgumentException here.